src/Illuminate/Testing/Concerns/TestDatabases.php

```diff
@@ -146,22 +146,24 @@ protected function whenNotUsingInMemoryDatabase($callback)
      */
     protected function switchToDatabase($database)
     {
-        DB::purge();
-
-        $default = config('database.default');
-
-        $url = config("database.connections.{$default}.url");
-
-        if ($url) {
-            config()->set(
-                "database.connections.{$default}.url",
-                preg_replace('/^(.*)(\/[\w-]*)(\??.*)$/', "$1/{$database}$3", $url),
-            );
-        } else {
-            config()->set(
-                "database.connections.{$default}.database",
-                $database,
-            );
+        $connections = config('database.testing.connections', [config('database.default')]);
```

Comment:
Something like `database.testing` does not exist?

Not sure if it's feasible to to something like https://github.com/laravel/framework/blob/a32a32f628eec0cee87d4e7d4b32891cfb45b2e4/src/Illuminate/Foundation/Testing/DatabaseTransactions.php#L16
which effectively makes use of a property `connectionsToTransact`

Reply:
I was under the impression that we don't have access to the test case instance. I'm updating this PR with a cleaner implementation.